packages/vitest/src/runtime/run.ts

```diff
@@ -27,6 +27,20 @@ export async function runTest(test: Test) {
   if (test.mode !== 'run')
     return
 
+  const regex = new RegExp(process.__vitest_worker__.ctx.config.testNamePattern)
+  if (!regex.test(test.name)) {
+    const startAndEnd = performance.now()
+    test.result = {
```

Comment:
it might also be better to directly set `mode = 'skip'` maybe in here
https://github.com/vitest-dev/vitest/blob/e7a85d0003f382cf9c4584d71bada72e8f94cf2b/packages/vitest/src/utils/index.ts#L59